src/ol_infrastructure/substructure/aws/eks/__main__.py

```diff
@@ -0,0 +1,291 @@
+)
+vault_k8s_auth_backend_role = vault.kubernetes.AuthBackendRole(
+    f"{cluster_name}-eks-vault-authentication-endpoint-operations",
+    role_name="operations-default",
+    backend=cluster_stack.require_output("vault_auth_endpoint"),
+    bound_service_account_names=[vault_traefik_service_account_name],
+    bound_service_account_namespaces=["operations"],
+    token_policies=[vault_traefik_policy_name],
+)
+# Create a k8s service account that will make the requests to vault
+vault_traefik_service_account = kubernetes.core.v1.ServiceAccount(
+    f"{cluster_name}-vault-traefik-service-account",
+    metadata=kubernetes.meta.v1.ObjectMetaArgs(
+        name=vault_traefik_service_account_name,
+        labels=k8s_global_labels,
+        namespace="operations",
+    ),
+    automount_service_account_token=False,
+    opts=ResourceOptions(
+        provider=k8s_provider,
+        parent=k8s_provider,
+        delete_before_replace=True,
+    ),
+)
+
+# We need to give the ServiceAccount the 'system:auth-delegator' ClusterRole
+# which will allow vault to use the token that the request it makes to turn around
+# and validate the request from the cluster api endpoint.
+#
+# Every application that wishes to use the vault-secrets-operator
+# will need to implement this pattern because ServiceAccounts don't
+# come with long-lived tokens starting with k8s 1.21
+#
+# This operational overhead is annoying but it is the best / most secure option
+# for when working with a vault installation that resides outside of the cluster
+#
+# Ref: https://developer.hashicorp.com/vault/docs/auth/kubernetes#use-the-vault-client-s-jwt-as-the-reviewer-jwt
+# Ref: https://developer.hashicorp.com/vault/docs/auth/kubernetes#configuring-kubernetes
+vault_traefik_service_account_cluster_role_binding = (
+    kubernetes.rbac.v1.ClusterRoleBinding(
+        f"{cluster_name}-vault-traefik-service-account-cluster-role-binding",
+        args=kubernetes.rbac.v1.ClusterRoleBindingInitArgs(
+            metadata=kubernetes.meta.v1.ObjectMetaArgs(
+                name=f"{vault_traefik_service_account_name}:cluster-auth",
+                labels=k8s_global_labels,
+                namespace="operations",
+            ),
+            role_ref=kubernetes.rbac.v1.RoleRefArgs(
+                api_group="rbac.authorization.k8s.io",
+                kind="ClusterRole",
+                name="system:auth-delegator",
+            ),
+            subjects=[
+                kubernetes.rbac.v1.SubjectArgs(
+                    kind="ServiceAccount",
+                    name=vault_traefik_service_account_name,
+                    namespace="operations",
+                ),
+            ],
+        ),
+        opts=ResourceOptions(
+            provider=k8s_provider,
+            parent=vault_traefik_service_account,
+            delete_before_replace=True,
+        ),
+    )
+)
+
+# Create a VaultAuth resource and a VaultStaticSecret resource
```

Comment:
Does this mean that we'll need a VaultStaticSecret per namespace per secret?

Reply:
Per Namespace / application / logical use case :
1 VaultConnection Resource (the helm chart makes the one we're using here)
1 ServiceAccount Resource
1 ClusterRoleBinding Resource
1 VaultAuth Resource

Per Secret:
X VaultDynamicSecret resources and/or Y VaultStaticSecret resources based on requirements, each of which maps to a k8s Secret Resource.